src/main/java/OOP/Parser.java

```diff
@@ -0,0 +1,97 @@
+                                throw new DukeException("\tEmpty Description");
+                            } else if (twoParts.length != 2) {
+                                throw new DukeException("\tUsage: deadline {taskName} /by {yyyy-MM-dd HHmm}");
+                            }
+                            String deadlineString = twoParts[1];
+                            return new AddDeadlineCommand(deadlineName, deadlineString);
+                        case "event":
+                            String[] threeParts = userCommandText.split ("/");
+                            String eventName = extractSecondWordOnwards(threeParts[0]);
+                            if (eventName.length() == 0) {
+                                throw new DukeException("\tEmpty Description");
+                            }
+                            if (threeParts.length != 3) {
+                                throw new DukeException("\tIncorrect format for event." +
+                                                        "\n\tExpected usage: " +
+                                                        "event {eventName} /from {eventStart} /to {eventEnd}");
+                            }
+                            String eventStart = extractSecondWordOnwards(threeParts[1]);
+                            String eventEnd = extractSecondWordOnwards(threeParts[2]);
+                            if (eventStart.length() == 0 || eventEnd.length() == 0) {
+                                throw new DukeException("\tBoth event start and end date times must be specified.");
+                            }
+                            return new AddEventCommand(eventName, eventStart, eventEnd);
+                        case "mark":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: mark {id}");
+                            }
+                            return new MarkTaskCommand(id);
+                        case "unmark":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: unmark {id}");
+                            }
+                            return new UnmarkTaskCommand(id);
+                        case "delete":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: delete {id}");
+                            }
+                            return new DeleteTaskCommand(id);
+                        default:
+                            return new InvalidCommand();
```

Comment:
Should the case statements be indented according to Java coding standard introduced by the module?